src/sentry/integrations/base.py

```diff
@@ -324,6 +329,13 @@ def org_integration(self) -> RpcOrganizationIntegration | None:
     def org_integration(self, org_integration: RpcOrganizationIntegration) -> None:
         self._org_integration = org_integration
 
+    @cached_property
+    def organization(self) -> RpcOrganization:
+        organization = organization_service.get(id=self.organization_id)
+        if organization is None:
+            raise Exception("organization_id not found")
```

Comment:
Should this raise a `NotFoundException` so that the endpoint response with a 404 and not a 500?

Reply:
🤔 Seems like a *slight* violation of abstraction as this class isn't strictly part of the ReST layer, but it's probably better to do it than not.

For the record, the `Organization.objects.get_from_cache` query that this replaces would have raised an `Organization.DoesNotExist` exception and caused a 500 status, so it's not a covered failure mode. But I agree that it's worth future-proofing, especially since I'm setting it up for reuse in other `IntegrationInstallation` subclasses.